praw/reddit.py

```diff
@@ -272,9 +277,32 @@ def _prepare_objector(self):
         for kind, klass in mappings.items():
             self._objector.register(kind, klass)
 
-    def _prepare_prawcore(self):
-        requestor = Requestor(USER_AGENT_FORMAT.format(self.config.user_agent),
-                              self.config.oauth_url, self.config.reddit_url)
+    def _prepare_prawcore(self, requestor_class=None, requestor_kwargs=None):
+        requestor_class = requestor_class or Requestor
+        requestor_kwargs = requestor_kwargs or {}
+
+        # Allow overriding oauth_url and reddit_url as Requestor keywords
+        oauth_url = requestor_kwargs.pop('oauth_url', None)
+        oauth_url = oauth_url or self.config.oauth_url
+        reddit_url = requestor_kwargs.pop('reddit_url', None)
+        reddit_url = reddit_url or self.config.reddit_url
+
+        # Example usage:
```

Comment:
I was thinking it would be great to make this part of the docstring for `Reddit` so that it shows up in the documentation. See examples for the `.. code:: python` for example here: https://github.com/praw-dev/praw/blob/d93d51c135432d6824793305b25c2cc64c0c0dc6/praw/models/reddit/submission.py#L51

I'm not sure why the ones in this file say `.. code-block`, maybe their aliases.